ci: Test and build project (#85)
MobilityData/gtfs-realtime-validator: DOWNLOAD_SNAPSHOT_JAR.md

> # Instructions to download the latest snapshot version of the validator

Reviewer: This is a lot of steps to follow to get the latest SNAPSHOT, including some manual steps that are error prone (with two actions having the same name).

Previously we used Travis to build a ` gtfs-realtime-validator-lib-v1.0.0-SNAPSHOT` and ` gtfs-realtime-validator-webapp-v1.0.0-SNAPSHOT` and push them to a public Amazon S3 bucket on each commit (this predated GitHub Actions). This let us put a stable URL on the README where someone could always access the latest SNAPSHOT with a single click.

Can we do something similar here to have a single click to get the latest SNAPSHOT? Is there a GitHub shortcut/redirect to the latest run or artifact for a particular action?

Having the hash and branch name in the JAR name is nice as it makes it simple to tell it's origin, but it definitely makes the above more complicated.

Reviewer 2: This is definitely something we could do. Should it be added as a nice to have in another issue/PR? The process described here is the same as the GTFS Schedule validator one. What are your thoughts on this @isabelle-dr ? Either works for me.

Reviewer 3: Hi Maxime. Thank you for working on this 🙏. 
I'm in favour of using the JAR from the latest stable release, not the master branch.
We do this in the Schedule Validator by providing a direct link to past releases in the README, and we update the README from the release with a direct link to the JAR (see below).
We just need to make sure we have releases regularly so the users aren't missing out on a functional newer version of the validator.

<img width="500" alt="Screen Shot 2022-02-17 at 9 03 36 AM" src="https://user-images.githubusercontent.com/63653518/154497563-6c9f502c-379a-4cae-86be-331cb7081e3f.png">

That being said, having a stable URL to the latest SNAPSHOT with a single click would be useful. I'd say we can open an issue and prioritize this modification according to the feedback we receive from users.

Reviewer 3: No, this is good :)

> The following workflows produce artifacts that are stored for a period of 90 days:
> 
> More particularly, [`test_pack_doc.yml`](../.github/workflows/test_package.yml) is responsible to automatically test and package the project.
> This workflow is triggered on each commit of each branch. 

Reviewer: ```suggestion
The GitHub Action [`test_package.yml`](../.github/workflows/test_package.yml) is responsible for automatically testing and packaging the project on each commit of each branch. Artifacts are stored for 90 days.
```

As a result, the latest snapshot version of the validator can be found under the list of artifacts generated when this workflow is executed on the `master` branch.
The application jar is named as "gtfs-realtime-validator-lib-v-*-SNAPSHOT.jar".

> This also applies to the latest webapp snapshot version, for which the webapp jar is named as "gtfs-realtime-validator-webapp-v-*-SNAPSHOT.jar".

Reviewer: The latest snapshot builds of the validator library and webapp can be found under the [list of artifacts for the master branch](https://github.com/MobilityData/gtfs-realtime-validator/actions/workflows/test_package.yml?query=branch%3Amaster):
* Library - `gtfs-realtime-validator-lib-v-*-SNAPSHOT.jar`
* Webapp - `gtfs-realtime-validator-webapp-v-*-SNAPSHOT.jar`

...where `*` is the branch and SHA1 commit of the build.

See the following instructions to download the artifact:

1. Access the actions listing on the project's main page
1. Select `Test Package` in the `Workflows` column
1. Select `master` branch
![access actions](https://user-images.githubusercontent.com/46797220/154085290-89666a47-4825-465d-9b56-83331b5aa363.png)
These three aforementioned steps can be skipped if you directly go to the following url: [Test Package workflow executions on master branch](https://github.com/MobilityData/gtfs-realtime-validator/actions/workflows/test_package.yml?query=branch%3Amaster)

1. Select the first item in the list: it is the latest iteration of the workflow that was run on the master branch

> 1. Click on the artifact's name that is needed to start download 

Reviewer: In https://github.com/MobilityData/gtfs-realtime-validator/actions/runs/1847621131, when I click on the artifact name (e.g., `gtfs-realtime-validator-webapp-v-pr-85-sha-1f238226-SNAPSHOT.jar`), it downloads the file `gtfs-realtime-validator-lib-v-pr-85-sha-1f238226-SNAPSHOT.jar.zip`, which is a ZIP archive and not the raw JAR.

Do you see the same thing? Does GitHub do this automatically or can we have a raw JAR for everyone to download?

If it's a ZIP file, then we need another step here to tell people how to extract the JAR from the ZIP.

Reviewer 2: Oh right. It is done automatically by GitHub, so I will add a step for the JAR extraction. This file was taken and updated from the gtfs-validator repo, meaning we should update it there as well.

# Instructions to download the latest snapshot version of the validator

> The GitHub Action [`test_package.yml`](../.github/workflows/test_package.yml) is responsible for automatically deploying the project when a push is made to the `master` branch. The artifacts can be found on GitHub Packages: [GTFS Realtime Validator Packages](https://github.com/orgs/MobilityData/packages?repo_name=gtfs-realtime-validator).

Reviewer: Now that this page is significantly simplified I'd remove it entirely, and just put the links to the library and webapp packages in-line in the respectively sections of the README, just like they were before. Looks like you've added the links, and I don't think this file is referenced anywhere else, so you can delete it.
